src/resolvers/default.rs

```diff
@@ -504,20 +508,21 @@ impl Kem for Kyber1024 {
         shared_secret_out: &mut [u8],
         ciphertext_out: &mut [u8],
     ) -> Result<(usize, usize), ()> {
-        let pubkey = kyber1024::PublicKey::from_bytes(pubkey).map_err(|_| ())?;
-        let (shared_secret, ciphertext) = kyber1024::encapsulate(&pubkey);
-        shared_secret_out.copy_from_slice(shared_secret.as_bytes());
-        ciphertext_out.copy_from_slice(ciphertext.as_bytes());
-        Ok((shared_secret.as_bytes().len(), ciphertext.as_bytes().len()))
+        let pk = self.alg.public_key_from_bytes(pubkey).ok_or_else(|| ())?;
+        self.alg.encapsulate(&pk).map_err(|_| ())?;
+        let (ciphertext, shared_secret) = self.alg.encapsulate(&pk).map_err(|_| ())?;
+        shared_secret_out.copy_from_slice(&shared_secret.into_vec());
```

Comment:
Looks like `Ciphertext` and `SharedSecret` implement `AsRef<[u8]>` so it seems like the `into_vec()` is unnecessary.

Reply:
yes i think you're right.